### PR TITLE
Notify users on group membership addition and deletion

### PR DESCRIPTION
**Pull request by srbaker**

### CODE REVIEW

File: src/api/app/models/event/added_user_to_group.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module Event
+  class AddedUserToGroup < Base
+    self.message_bus_routing_key = 'groups_user.create'
```
Comment (hennevogel) [inline src/api/app/models/event/added_user_to_group.rb:3]: I'm don't think this should go over the message bus. It's a personal notification, not a public one.

File: src/api/app/models/event/removed_user_from_group.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+module Event
+  class RemovedUserFromGroup < Base
+    self.message_bus_routing_key = 'groups_user.delete'
```
Comment (hennevogel) [inline src/api/app/models/event/removed_user_from_group.rb:3]: Same here, no need to route this over the message bus.

File: src/api/spec/models/groups_user_spec.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+RSpec.describe GroupsUser do
+  let(:group) { create(:group) }
+  let(:user) { create(:confirmed_user, login: 'eisendieter') }
+
+  it 'creates an AddedToGroupEvent when a user is added to a group' do
+    allow(Event::AddedUserToGroup).to receive(:create)
+
+    group.add_user(user)
+
+    expect(Event::AddedUserToGroup).to have_received(:create).with(group: group, user: user)
```
Comment (hennevogel) [inline src/api/spec/models/groups_user_spec.rb:10]: Hm, how do you expect this to break? 🤔

**Review by hennevogel — CHANGES_REQUESTED** [review]

**Comment by saraycp** [comment]
Could you please remove the commits that don't apply any changes in the end? I mean the commits that revert changes and the corresponding commits which introduced those changes.

File: src/api/app/models/event/added_user_to_group.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+    self.description = 'Added user to group'
+    payload_keys :group, :user
```
Comment (hennevogel) [inline src/api/app/models/event/added_user_to_group.rb:5]: I'm also missing `who` did that change here...

File: src/api/app/models/event/added_user_to_group.rb (hunk 1 above)
Comment (srbaker) [reply src/api/app/models/event/added_user_to_group.rb:3]: Aha!  I'm slightly unaware of this; I copied it from an example I was looking at.  Is there documentation on the difference that I missed?

File: src/api/spec/models/groups_user_spec.rb (hunk 3 above)
Comment (srbaker) [reply src/api/spec/models/groups_user_spec.rb:10]: Well, the call is behaviour.  And the call is made from `#add_user`.  The mock is about testing the edges, not the nodes.

**Comment by srbaker** [comment]
> Could you please remove the commits that don't apply any changes in the end? I mean the commits that revert changes and the corresponding commits which introduced those changes.

Aiming to do that now.  I am terrible at this part.

File: src/api/app/models/event/added_user_to_group.rb (hunk 4 above)
Comment (srbaker) [reply src/api/app/models/event/added_user_to_group.rb:5]: Aha.  I didn't see any copy for the messages;  this was basically a placeholder at the time.

**Comment by srbaker** [comment]
> This PR implements the event part. I tried manually and the event is created correctly when someone adds or removes a user to/from a group.
> 
> If I'm not wrong, the only missing part to finish the event part is defining `receiver_roles`. I can help out.

This part I didn't quite understand.  I'm going to do some fixes to cleanup bits above but if you can do this part I would appreciate it.

File: src/api/app/models/event/added_user_to_group.rb (hunk 1 above)
Comment (hennevogel) [reply src/api/app/models/event/added_user_to_group.rb:3]: https://github.com/openSUSE/open-build-service/wiki/Events#publishing-events-to-the-rabbitmq-message-bus

**Comment by saraycp** [comment]
Close this PR in favor of PR #16242

**Pull request closed without merging**